Las matrices A y B representan las ventas en dos años, en millones de dólares, de los tres productos de una empresa en las cuatro regiones en que está disponible. Es decir, a_{ij} y b_{ij} representan el monto de las ventas del producto i en la región j el primer y segundo año, respectivamente:

<br>

\begin{split}
A = \begin{pmatrix}
 2,6 &  4,8 &  1,8 & 0,9 \\
 3,2 &  4,4 &  2,5 & 2,8 \\
 2,4 &  3,6 &  3,8 & 2,5 \\
\end{pmatrix} \text{ } B= \begin{pmatrix}
 3,6 &  2,5 &  3,0 & 2,5 \\
 4,5 &  5,0 &  3,5 & 3,8 \\
 2,9 &  3,0 &  4,6 & 4,0 \\
\end{pmatrix}
\end{split}

<br>

1. Indique el orden de cada matriz.
2. Calcule la matriz S = A + B.
3. Calcule la matriz D = B - A.
4. Indique el valor e interprete los elementos s_{24} y d_{24}.
5. Para el tercer año de funcionamiento, los tres productos ahora están disponibles en seis regiones. Durante este año, las ventas de esta empresa estarán registradas en una matriz E. ¿Es posible determinar la matriz T = A + B + E? Justifique su respuesta.

In [ ]:
import numpy as np 
A=np.array([[2.6,4.8,1.8,0.9],
            [3.2,4.4,2.5,2.8],
            [2.4,3.6,3.8,2.5]])

B=np.array([[3.6,2.5,3.0,2.5],
            [4.5,5.0,3.5,3.8],
            [2.9,3.0,4.6,4.0]])

S=A+B
D=B-A
print("1°)\n    Cada matriz es de orden: 3*4")
print("2°)\n",S)
print("\n3°)\n",D)
print("\n4°)\nPara la matriz S:\n   El producto 2 en la región 4 se vendió 6.6 millones de dólares en un lapso de 2 años")
print("Para la matriz D:\n  La diferencia entre ambos años indica que en el segundo año el producto 2 en la region 4 aumentó 1 millón en ventas")
print("5°)\n    No se puede, puesto que para sumar todas las matrices deben tener el mismo orden y la matriz E es de orden 3*6")

2°)
 [[6.2 7.3 4.8 3.4]
 [7.7 9.4 6.  6.6]
 [5.3 6.6 8.4 6.5]]

3°)
 [[ 1.  -2.3  1.2  1.6]
 [ 1.3  0.6  1.   1. ]
 [ 0.5 -0.6  0.8  1.5]]

4°)
Para la matriz S:
   El producto 2 en la región 4 se vendió 6.6 millones de dólares en un lapso de 2 años
Para la matriz D:
  La diferencia entre ambos años indica que en el segundo año el producto 2 en la region 4 aumentó 1 millón en ventas
5°)
    No se puede, puesto que para sumar todas las matrices deben tener el mismo orden y la matriz E es de orden 3*6


# Problema 2

Una fábrica con sucursales en Japón y China, produce tres tipos de componentes $j$, para cinco modelos de computador $i$. Los costos de producción mensual, en miles de dólares, de ambas sucursales se representan en las matrices $J$ y $C$, respectivamente:

<br>

\begin{split}
J = \begin{pmatrix}
 100 &  200 &  150  \\
 120 &  210 &  170  \\
 112 &  200 &  131  \\
 124 &  310 &  170  \\
 136 &  156 &  140  \\
\end{pmatrix} \text{ } C = \begin{pmatrix}
 110 &  190 &  160  \\
 130 &  220 &  180  \\
 146 &  190 &  141  \\
 143 &  312 &  163  \\
 137 &  116 &  156  \\
\end{pmatrix}
\end{split}

<br>

1. ¿Cuál es el orden de las matrices?
2. Inteprete el elemento $j_{21}$ y $c_{12}$ en el contexto del problema.
3. Utilizando *Python*, calcule el costo total de producción para cada modelo, considerando ambas sucursales.
4. Un trabajador indica que, en la sucursal japonesa, el costo de producción del producto $3$, modelo $4$ de computador, es más alto que en la sucursal china. ¿Tiene razón? Justifique apropiadamente.

In [11]:
import numpy as np 
print(f"1°) La matriz J y C son de orden: 5 * 3 \n")
print("2°) Para Japón:\n  El primer componente del segundo modelo cuesta 120 mil dólares en fabricarse\n  Mientras que para China:\n  El segundo componente del primer modelo tiene un costo de 190 mil dólares mensuales")



# Definir las matrices J y C
J = np.array([
    [100, 200, 150],
    [120, 210, 170],
    [112, 200, 131],
    [124, 310, 170],
    [136, 156, 140]
])

C = np.array([
    [110, 190, 160],
    [130, 220, 180],
    [146, 190, 141],
    [143, 312, 163],
    [137, 116, 156]
])

# Sumar matrices
S = J + C
print("\n3°) Suma de J y C:        \n",S)
print("\n4°) Sí, tiene razón el empleado:\nEsto se debe a que en el contexto del problema se representan los modelos * componente, en este caso son de orden 5*3")
print("Entendiendo el contexto y orden sabemos que la pregunta corresponde a la fila 4 y columna 3 en cada matriz, cuyos valores son: ")
print(f"Japón: {J[3][2]} mil y China: {C[3][2]} mil\nComprobando que es correcta la afirmación del trabajador ")




1°) La matriz J y C son de orden: 5 * 3 

2°) Para Japón:
  El primer componente del segundo modelo cuesta 120 mil dólares en fabricarse
  Mientras que para China:
  El segundo componente del primer modelo tiene un costo de 190 mil dólares mensuales

3°) Suma de J y C:        
 [[210 390 310]
 [250 430 350]
 [258 390 272]
 [267 622 333]
 [273 272 296]]

4°) Sí, tiene razón el empleado:
Esto se debe a que en el contexto del problema se representan los modelos * componente, en este caso son de orden 5*3
Entendiendo el contexto y orden sabemos que la pregunta corresponde a la fila 4 y columna 3 en cada matriz, cuyos valores son: 
Japón: 170 mil y China: 163 mil
Comprobando que es correcta la afirmación del trabajador 


# Problema 3

La empresa Beta tiene cinco plantas en Chile, y en cada una de ellas fabrica siete productos. El número de unidades del producto $i$ fabricadas en la planta $j$ en una semana está representado en la matriz $C$ en el elemento $c_{ij}$.

<br>

\begin{split}
C = \begin{pmatrix}
 120 &  200 &  155 &  125  &  230 \\
 155 &  130 &  110 &  160  &  180 \\
 140 &  205 &  150 &  125  &  155 \\
 130 &  120 &  230 &  225  &  125 \\
 135 &  70  &  190 &  175  &  95  \\
 110 &  265 &  180 &  230  &  180 \\
 185 &  85  &  120 &  135  &  150 \\
\end{pmatrix}
\end{split}

<br>

La empresa se expande, creando cinco plantas en Ecuador que producen los mismos siete productos. La producción semanal de esta filial estará representada por una matriz $E$.

1. Si se sabe que los volúmenes de producción semanal de la filial en Ecuador son un $20\%$ menor que en Chile, calcule la matriz $E$ utilizando *Python*.
2. Si representamos la producción semanal total de la empresa Beta, considerando ambas filiales, por la matriz $T$, escriba la expresión que calcula cada elemento de la matriz $T$ usando notación matricial.
3. Calcule la matriz $T$ utilizando *Python*.
4. Interprete los elementos $c_{34}$ y $t_{72}$ en el contexto del problema.

# Problema 4

Una empresa que fabrica zapatos tiene dos plantas, una en Valparaíso y la otra en Temuco. La empresa produce zapatos de color negro, blanco y café, tanto para niños, damas como caballeros. La capacidad actual de producción mensual (en miles de pares) para Valparaíso y Temuco, están dadas por las matrices $V$ y $T$, respectivamente. Para ambas matrices, según el orden en que se mencionan, las filas corresponden al color del calzado y las columnas al tipo de persona que utilizará el calzado.

<br>

\begin{split}
V = \begin{pmatrix}
 30 &  34 &  20  \\
 45 &  20 &  16  \\
 14 &  26 &  25  \\
\end{pmatrix} \text{ } T = \begin{pmatrix}
 35 &  30 &  26  \\
 52 &  25 &  18  \\
 23 &  24 &  32  \\
\end{pmatrix}
\end{split}

<br>

1. Determine la producción mensual total de ambas plantas, representándola en una matriz $A$. Utilice *Python* para resolver.
2. Interprete el elemento $a_{21}$ en el contexto del problema.
3. Si para el próximo año se estima que la producción en Valparaíso aumenta en un $30\%$, mientras que la de Temuco se reduce en un $10\%$, represente la nueva producción total en la matriz $B$.
4. Interprete el elemento $b_{13}$ en el contexto del problema.

# Problema 5

Genere una matriz $M$ de orden $8 \times 9$ donde cada término de la matriz está definido por:

$$m_{ij} = 8i + 12j$$

y calcule su transpuesta.

In [2]:
import numpy as np 
M = np.fromfunction (lambda i,j: 8*(i+1)+12*(j+1), (8,9))
M_T = M.T
print(f"La matriz M es:\n{M}\n ")
print(f"La matriz transpuesta de M es:\n {M_T}:  ")



La matriz M es:
[[ 20.  32.  44.  56.  68.  80.  92. 104. 116.]
 [ 28.  40.  52.  64.  76.  88. 100. 112. 124.]
 [ 36.  48.  60.  72.  84.  96. 108. 120. 132.]
 [ 44.  56.  68.  80.  92. 104. 116. 128. 140.]
 [ 52.  64.  76.  88. 100. 112. 124. 136. 148.]
 [ 60.  72.  84.  96. 108. 120. 132. 144. 156.]
 [ 68.  80.  92. 104. 116. 128. 140. 152. 164.]
 [ 76.  88. 100. 112. 124. 136. 148. 160. 172.]]
 
La matriz transpuesta de M es:
 [[ 20.  28.  36.  44.  52.  60.  68.  76.]
 [ 32.  40.  48.  56.  64.  72.  80.  88.]
 [ 44.  52.  60.  68.  76.  84.  92. 100.]
 [ 56.  64.  72.  80.  88.  96. 104. 112.]
 [ 68.  76.  84.  92. 100. 108. 116. 124.]
 [ 80.  88.  96. 104. 112. 120. 128. 136.]
 [ 92. 100. 108. 116. 124. 132. 140. 148.]
 [104. 112. 120. 128. 136. 144. 152. 160.]
 [116. 124. 132. 140. 148. 156. 164. 172.]]:  


# Problema 6

Genere dos matrices:

*   Una matriz $A$ de orden $5 \times 7$, con $a_{ij} = i \cdot j.$
*   Una matriz $B$ de orden $7 \times 5$, con $b_{ij} = 15i - 2j.$

y calcule $C = A^{T} + \dfrac{1}{2}B$.



In [14]:
import numpy as np 
A = np.fromfunction (lambda i,j: (i+1)*(j+1),(5,7))
B = np.fromfunction(lambda i,j: 15*(i+1)-2*(j+1),(7,5))
print(f"Matriz A:\n{A}\n")
print(f"Matriz B:\n{B}\n")

A_T=A.T
BM=B/2
C=A_T+BM
print(f"Matriz C:\n{C}")

Matriz A:
[[ 1.  2.  3.  4.  5.  6.  7.]
 [ 2.  4.  6.  8. 10. 12. 14.]
 [ 3.  6.  9. 12. 15. 18. 21.]
 [ 4.  8. 12. 16. 20. 24. 28.]
 [ 5. 10. 15. 20. 25. 30. 35.]]

Matriz B:
[[ 13.  11.   9.   7.   5.]
 [ 28.  26.  24.  22.  20.]
 [ 43.  41.  39.  37.  35.]
 [ 58.  56.  54.  52.  50.]
 [ 73.  71.  69.  67.  65.]
 [ 88.  86.  84.  82.  80.]
 [103. 101.  99.  97.  95.]]

Matriz C:
[[ 7.5  7.5  7.5  7.5  7.5]
 [16.  17.  18.  19.  20. ]
 [24.5 26.5 28.5 30.5 32.5]
 [33.  36.  39.  42.  45. ]
 [41.5 45.5 49.5 53.5 57.5]
 [50.  55.  60.  65.  70. ]
 [58.5 64.5 70.5 76.5 82.5]]


# Problema 7

Una red de sensores recoge datos de temperatura ($°C$), humedad (porcentaje) y presión (hPa) en cinco días distintos (de lunes a viernes). Los datos medidos se muestran en la matriz $D$:

<br>

\begin{equation*}
\begin{matrix}
& \begin{matrix}
 & Lu  \ \ &Ma \ \ &Mi \ \ & Ju \ \ &Vi
\end{matrix} \\
&D = \begin{pmatrix}
20    &  28    &  22    &  26     &  24   \\
52    &  57    &  56    &  54     &  58   \\
1.066 &  1.068 &  1.066 &  1.064  &  1.068 \\
\end{pmatrix} & \ \begin{matrix}
Cº\\
\% \\
hPa
\end{matrix}\\
\end{matrix}
\end{equation*}

<br>

Se sabe que por un error de calibración de los sensores, las medidas registradas deberían ser un $5\%$ menores que las registradas.

1. Calcule en una matriz $S$ los valores correctos de temperatura, humedad y presión. Utilice *Python* para calcular.
2. Obtenga la matriz $S^{T}$.
3. Interprete los elementos $s_{24}$ y $s_{31}^{T}$.


# Problema 8

Una empresa tecnológica vende software y hardware en tres productos: $P_{1}$, $P_{2}$ y $P_{3}$, en cuatro zonas de la capital: norte, sur, este y oeste. Las ventas del mes, en millones de pesos, están registradas en las matrices $S$ y $H$, para software y hardware respectivamente.

<br>

\begin{split}
S = \begin{pmatrix}
 260 &  480 &  180  &  90  \\
 320 &  440 &  250  &  280 \\
 240 &  360 &  380  &  250  \\
\end{pmatrix} \text{ } H = \begin{pmatrix}
 360 &  250 &  300  &  250  \\
 450 &  500 &  350  &  380 \\
 290 &  300 &  460  &  400  \\
\end{pmatrix}
\end{split}

<br>

1. Calcule en la matriz $T$, del total de ventas de la empresa tecnológica.
2. Un trabajador indica que este mes, la sucursal de la zona sur recaudó más por la venta hardware que software, ¿es correcta su afirmación?
3. Debido al black friday se espera que el siguiente mes, las ventas de software aumenten un $7\%$ y las de hardware aumenten un $3\%$ respecto al mes actual. Encuentra en total de ventas esperadas para el siguiente mes y almacénalo en la matriz $M$. Utiliza *Python* para realizar los cálculos.
4. Interpreta los elementos $t_{24}$ y $m_{31}$ en el contexto del problema.
5. Calcula la matriz $M^{T}$ e interpreta el elemento $m_{41}^{T}$ en el contexto del problema.
